symbolic-debuginfo/src/pe.rs

```diff
@@ -318,14 +309,33 @@ impl<'data> PeObject<'data> {
 
                 // See data specification:
                 // https://github.com/dotnet/runtime/blob/97ddb55e3adde20ceac579d935cef83cfe996169/docs/design/specs/PE-COFF.md#embedded-portable-pdb-debug-directory-entry-type-17
-                offset += 4; // skip `signature`
+                // let signature = &self.data[offset..(offset + 4)];
+                let mut signature: [u8; 4] = [0; 4];
+                self.data
+                    .gread_inout(&mut offset, &mut signature)
+                    .map_err(PeError::new)?;
+                if signature != "MPDB".as_bytes() {
```

Comment:
```suggestion
                let signature: [u8; 4] = self.data
                    .gread(&mut offset)
                    .map_err(PeError::new)?;
                if signature != b"MPDB" {
```
First time I saw `gread_inout`.

Reply:
not the same thing - `gread` doesn't work with slices

Reply:
```
error[E0277]: the trait bound `[u8; 4]: TryFromCtx<'_, _>` is not satisfied
   --> symbolic-debuginfo\src\pe.rs:313:42
    |
313 |                 let signature: [u8; 4] = self.data.gread(&mut offset).map_err(PeError::new)?;
    |                                          ^^^^^^^^^ ----- required by a bound introduced by this call
    |                                          |
    |                                          the trait `TryFromCtx<'_, _>` is not implemented for `[u8; 4]`
    |
    = help: the trait `TryFromCtx<'a, usize>` is implemented for `&'a [u8]`
note: required by a bound in `gread`
   --> C:\Users\dlugo\.cargo\registry\src\github.com-1ecc6299db9ec823\scroll-0.11.0\src\pread.rs:88:21
    |
88  |     fn gread<'a, N: TryFromCtx<'a, Ctx, Self, Error = E>>(
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `gread`
    ```